lms/djangoapps/instructor/views/api.py

```diff
@@ -2629,8 +2636,20 @@ def spoc_gradebook(request, course_id):
         for student in enrolled_students
     ]
```

Comment:
I've discussed this feature with our ops team and they would like it so a
user only loads the number of students per page, rather than dumping all
students. We disable this feature on courses > 200 students, and it's still
a performance drain even on these small courses.

On Wed, Oct 28, 2015 at 7:57 AM, Amir Qayyum Khan <notifications@github.com>
wrote:

> In lms/djangoapps/instructor/views/api.py
> <https://github.com/edx/edx-platform/pull/10250#discussion_r43245817>:
>
> > @@ -2629,8 +2636,20 @@ def spoc_gradebook(request, course_id):
> >          for student in enrolled_students
> >      ]
>
> @pdpinch <https://github.com/pdpinch> django pagination api divides
> result into chunks that it call pages,
> My aim was to divide view into pages. So that user dont need to scroll a
> huge list.
> I have a question what is maximum number of students normally a edX course
> have?
> If you guys think student list will be huge then may be i need to change
> my logic here.
>
> Edx code example of pagination
> https://github.com/edx/edx-platform/blob/062e979e0ed14161d0f882091a06e032ca76bb3e/openedx/core/lib/api/paginators.py#L10
>
> https://docs.djangoproject.com/en/1.8/topics/pagination/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10250/files#r43245817>.
>